sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/windowing/WindowTest.java

```diff
@@ -220,6 +224,67 @@ public void testMissingLateness() {
       .apply("Trigger", Window.<String>triggering(trigger));
   }
 
+  @Test
+  @Category(RunnableOnService.class)
+  public void testNoWindowFnDoesNotReassignWindows() {
+    pipeline.enableAbandonedNodeEnforcement(true);
+
+    PCollection<Long> initialWindows =
+        pipeline
+            .apply(CountingInput.upTo(10L))
+            .apply(
+                "AssignInitialTimestamps",
+                WithTimestamps.of(
+                    new SerializableFunction<Long, Instant>() {
+                      @Override
+                      public Instant apply(Long input) {
+                        return new Instant(input);
+                      }
+                    }))
+            .apply("AssignWindows", Window.<Long>into(FixedWindows.of(Duration.millis(5L))));
+
+    PAssert.that(initialWindows)
```

Comment:
Comment that this is establishing an assumption, not the assertion that this test method is ensuring.